File: 线性表/844_比较含退格的字符串.py
class Solution:
    def getstring(self, s: str) -> str:
        out = []
        for char in s:
            if char != '#':
                out.append(char)
            elif out:
                out.pop()
        return ''.join(out)

    def backspaceCompare(self, S: str, T: str) -> bool:
        return self.getstring(S) == self.getstring(T)

File: 线性表/844_比较含退格的字符串.cpp
#include <stack>
#include <string>
using namespace std;

class Solution {
    public:
    bool backspaceCompare(string S, string T) {
        stack<char> s1, s2;
        auto        func = [](stack<char>& s, char ch) -> void {
            if (ch != '#') {
                s.push(ch);
            } else if (!s.empty()) {
                s.pop();
            }
        };

        for (auto c : S) {
            func(s1, c);
        }
        for (auto c : T) {
            func(s2, c);
        }
        return s1 == s2;
    }
};

int main() {
    Solution s;
    auto     ch = s.backspaceCompare("y#fo##f", "y#f#o##f");
}